kratos/utilities/sensitivity_builder.cpp

```diff
@@ -449,6 +450,94 @@ std::vector<std::string> ArrayOfStringToVector(Parameters Settings)
     return result;
 }
 
+template<class TContainerType>
+TContainerType& GetContainer(ModelPart& rModelPart);
+
+void AddMatrixSubBlock(
+    Matrix& rOutput,
+    const Matrix& rInput,
+    const int RowOffset,
```

Comment:
then there will be casts everywhere..., I think the neighbour id maps given by the `FindGlobalNodalNeighbours::GetNeighbourIds` should be unsigned, but if I change it, it might break existing code or give lots of warnings for everyone. I am not sure what is the correct way of doing it.